test/parameter-names-convention.test.js

```diff
@@ -33,6 +33,18 @@ test('az-parameter-names-convention should find errors', () => {
             type: 'string',
             description: 'Camel case header',
           },
+          {
+            name: '$fooBar',
```

Comment:
This name is wrong for multiple reasons.  Headers must be kebab-case, not camel case as here. So to make the test clear that it should trigger _only_ because of the leading '$', you should make the rest of the name kebab-case.
```suggestion
            name: '$foo-bar',
```